internal/config/config.go

```diff
@@ -470,6 +487,40 @@ func addressPoolFromCR(p metallbv1beta1.IPAddressPool) (*Pool, error) {
 		ret.cidrsPerAddresses[cidr] = nets
 	}
 
+	if p.Spec.AllocateTo != nil &&
```

Comment:
Let's wrap in a function that returns the ServiceAllocation.
Also, return early if allocateTo is nil